pkg/server/devfile-handler.go

```diff
@@ -0,0 +1,181 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/console/pkg/serverutils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog"
+
+	devfilev1 "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	devfile "github.com/devfile/library/pkg/devfile"
+	"github.com/devfile/library/pkg/devfile/generator"
+	"github.com/devfile/library/pkg/devfile/parser"
+)
+
+const (
+	dockerfilePath = "Dockerfile"
+)
+
+func (s *Server) devfileHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		data       devfileForm
+		devfileObj parser.DevfileObj
+	)
+
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to decode response: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	// Get devfile content and parse it using a library call in the future
+	devfileContentBytes := []byte(data.Devfile.DevfileContent)
+	devfileObj, err = devfile.ParseFromDataAndValidate(devfileContentBytes)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to parse devfile: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	containerComponents := devfileObj.Data.GetDevfileContainerComponents() //TODO: filter thru Console attribute, right now if there is more than one component container err out
+	if len(containerComponents) != 1 {
+		errMsg := fmt.Sprintf("Console Devfile Import Dev Preview, supports only one component container")
```

Comment:
You don't need `Sprintf` here since there are no parameters.

Reply:
TIL 🙂